application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,136 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Role;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.components.Button;
+import net.dv8tion.jda.api.interactions.components.ButtonStyle;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class WhoIsCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(WhoIsCommand.class);
+
+    private static final String USER_OPTION = "user";
+    private static final String HIDE_MEMBER_INFO = "hide_member_info";
+    private static final DateTimeFormatter DATE_TIME_FORMAT =
+            DateTimeFormatter.ofPattern("E, MMMM d, u, HH:mm:ss");
+
+    public WhoIsCommand() {
+        super("whois", "who is the given user?", SlashCommandVisibility.GUILD);
+
+        getData().addOption(OptionType.USER, USER_OPTION, "the user to look up", true);
+        getData().addOption(OptionType.BOOLEAN, HIDE_MEMBER_INFO,
+                "whenever to show the member info of this user", false);
```

Comment:
`"whether to hide information about this user that is specific to this server, such as when they joined"`

Reply:
edited